src/aleph/http/client.clj

```diff
@@ -59,19 +60,20 @@
                (URI.
                  (name (or (:scheme req) :http))
                  nil
-                 (or (:host req) (:server-name req))
+                 (when-some [host (or (:host req) (:server-name req))]
```

Comment:
A (subjective) matter of taste, but `some->` might be easier to read here. Will let the author of Elements of Clojure make the call though :)